docs/src/main/sphinx/connector/hive.rst

```diff
@@ -778,6 +778,10 @@ connector.
       - The target number of buffered splits for each table scan in a query,
         before the scheduler tries to pause.
       - ``1000``
+    * - ``hive.max-outstanding-splits-size``
+      - The maxiumum size allowed for buffered splits for each table scan
+        in a query, before the query fails.
```

Comment:
hm .. why is this causing a fail when the above property just causes a pause

Reply:
@alexjo2144 Can you confirm if this will cause a fail vs a pause?

Reply:
It indeed causes failure. The reason is that there is already `max-outstanding-splits` which causes scheduler to pause. Even if with that limit we hit this condition it suggests splits are very very large for some reason.

e.g. If max-outstanding-splits is 10,000 and each split is 1kB it gets us 10MB of mem usage.
If however each split is 1MB it gets us 10GB of mem usage. The problem is that in current implementation it's not possible to "pause" based on memory usage, only count of splits so the only option in this case is to fail the query.